hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/ProxyWorldUpdater.java

```diff
@@ -441,6 +448,22 @@ public void commit() {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setupContext(@NonNull final OperationContext context) {
```

Comment:
Is there a reason not to just inject the `HandleContext` into the `RootProxyWorldUpdater`?

Reply:
After the realization that this check is only needed for txs, there isn't. I updated the implementation